pkg/controller/constrainttemplate/constrainttemplate_controller.go

```diff
@@ -751,3 +773,104 @@ func shouldGenerateVAP(ct *templates.ConstraintTemplate, generateVAPDefault bool
 	}
 	return *source.GenerateVAP, nil
 }
+
+func vapForVersion(gvk *schema.GroupVersion) (client.Object, error) {
+	switch gvk.Version {
+	case "v1":
+		return &admissionregistrationv1.ValidatingAdmissionPolicy{}, nil
+	case "v1beta1":
+		return &admissionregistrationv1beta1.ValidatingAdmissionPolicy{}, nil
+	default:
+		return nil, errors.New("unrecognized version")
+	}
+}
+
+func (r *ReconcileConstraintTemplate) getRunTimeVAP(gvk *schema.GroupVersion, transformedVap *admissionregistrationv1beta1.ValidatingAdmissionPolicy, currentVap client.Object) (client.Object, error) {
+	if currentVap == nil {
+		if gvk.Version == "v1" {
+			return v1beta1ToV1(transformedVap)
+		}
+		return transformedVap.DeepCopy(), nil
+	}
+
+	if gvk.Version == "v1" {
+		v1CurrentVAP, ok := currentVap.(*admissionregistrationv1.ValidatingAdmissionPolicy)
+		if !ok {
+			return nil, errors.New("Unable to convert to v1 VAP")
+		}
+		v1CurrentVAP = v1CurrentVAP.DeepCopy()
+		tempVAP, err := v1beta1ToV1(transformedVap)
+		if err != nil {
+			return nil, err
+		}
+		v1CurrentVAP.Spec = tempVAP.Spec
+		return v1CurrentVAP, nil
+	}
+
+	v1beta1VAP, ok := currentVap.(*admissionregistrationv1beta1.ValidatingAdmissionPolicy)
+	if !ok {
+		return nil, errors.New("Unable to convert to v1 VAP")
+	}
+	v1beta1VAP.Spec = transformedVap.Spec
+	return v1beta1VAP.DeepCopy(), nil
+}
+
+func v1beta1ToV1(v1beta1Obj *admissionregistrationv1beta1.ValidatingAdmissionPolicy) (*admissionregistrationv1.ValidatingAdmissionPolicy, error) {
```

Comment:
Can we not use scheme.Convert here?

Reply:
Direct conversion from `v1beta1.ValidatingAdmissionPolicy` to `v1.ValidatingAdmissionPolicy` using `r.scheme.Convert` results in error - 
`converting (v1beta1.ValidatingAdmissionPolicy) to (v1.ValidatingAdmissionPolicy): unknown conversion`.

Reply:
Oof, sounds like a bug in K8s code. Interim solution LGTM

Reply:
@JaydipGabani do you want to open an issue for this on k/k?

Reply:
Here is the issue - https://github.com/kubernetes/kubernetes/issues/126582